database/sqlite3.ksy

```diff
@@ -32,213 +32,431 @@ doc: |
+      cases:
+        0: lock_byte_page(_index + 1)
+        1: ptrmap_page(_index + 1)
+        # TODO: Free pages and cell overflow pages are incorrectly interpreted as btree pages
+        # This is unfortunate, but unavoidable since there's no way to recognize these types at
+        # this point in the parser.
+        2: btree_page(_index + 1)
+    pos: 0
+    size: header.page_size
+    repeat: expr
+    repeat-expr: header.page_count
 types:
+  database_header:
+    seq:
+      - id: magic
+        contents: ["SQLite format 3", 0]
+      - id: page_size_raw
+        type: u2
+        doc: |
+          The database page size in bytes. Must be a power of two between
+          512 and 32768 inclusive, or the value 1 representing a page size
+          of 65536. The interpreted value is available as `page_size`.
+      - id: write_version
+        type: u1
+        enum: format_version
+        doc: File format write version
+      - id: read_version
+        type: u1
+        enum: format_version
+        doc: File format read version
+      - id: reserved_space
+        type: u1
+        doc: Bytes of unused "reserved" space at the end of each page. Usually 0.
+      - id: max_payload_fraction
+        type: u1
+        doc: Maximum embedded payload fraction. Must be 64.
+      - id: min_payload_fraction
+        type: u1
+        doc: Minimum embedded payload fraction. Must be 32.
+      - id: leaf_payload_fraction
+        type: u1
+        doc: Leaf payload fraction. Must be 32.
+      - id: file_change_counter
+        type: u4
+      - id: page_count
+        type: u4
+        doc: Size of the database file in pages. The "in-header database size".
+      - id: first_freelist_trunk_page
+        type: freelist_trunk_page_pointer
+        doc: Page number of the first freelist trunk page.
+      - id: freelist_page_count
+        type: u4
+        doc: Total number of freelist pages.
+      - id: schema_cookie
+        type: u4
+      - id: schema_format
+        type: u4
+        doc: The schema format number. Supported schema formats are 1, 2, 3, and 4.
+      - id: def_page_cache_size
+        type: u4
+        doc: Default page cache size.
+      - id: largest_root_page
+        type: u4
+        doc: The page number of the largest root b-tree page when in auto-vacuum or incremental-vacuum modes, or zero otherwise.
+      - id: text_encoding
+        type: u4
+        doc: The database text encoding. A value of 1 means UTF-8. A value of 2 means UTF-16le. A value of 3 means UTF-16be.
+      - id: user_version
+        type: u4
+        doc: The "user version" as read and set by the user_version pragma.
+      - id: is_incremental_vacuum
+        type: u4
+        doc: True (non-zero) for incremental-vacuum mode. False (zero) otherwise.
+      - id: application_id
+        type: u4
+        doc: The "Application ID" set by PRAGMA application_id.
+      - id: reserved
+        size: 20
```

Comment:
```diff
-db.header.reserved
+db.header.reserved_header_bytes
 db.header.reserved_space
```

?

Reply:
Sure. I changed
`reserved_space` -> `reserved_bytes_per_page`
`reserved` -> `reserved_header_bytes`
to make it clearer what these two value are.